components/DateRangeInput/index.jsx

```diff
@@ -0,0 +1,31 @@
+import React from "react"
+import moment from "moment"
+import classnames from "classnames"
+import { DatePicker, Props as DatePickerProps } from "../"
+
+import styles from "./styles.scss"
+
+export type Props = {
+  ...DatePickerProps,
+  fromDate: Object,
+  toDate: Object,
+}
+
+const DateRangeInput = ({ className, fromDate, toDate, isValidDate, viewDate, ...props }) => {
```

Comment:
1. How is this going to replace `DOBInput`?

It seems that one has to always provide a date range in order to use this component.
The DOB use case isn't necessarily a range, e.g. as long as the input value is a valid date prior to 18 years ago it should be valid.

I guess we could build a `DOBInput` on top of `DateRangeInput` where `fromDate` is old enough to include every reasonable possibility of a senior applicant (120 years old seems reasonable) and `toDate` would default to at least 18 years old?

2. How would this be used on the ID use case, issue vs expiration date?

`DateRangeInput` is a single input field. For this use case, I don't think we need a range per input (issue and expiration dates) but rather a valid range across inputs, e.g. as long as the `IssueDateInput` is in the past, and `ExpirationDateInput` is in the future, the range is valid.

Did I get that right?

Reply:
I guess my initial thought was to have components we can reuse everywhere in Nitro almost like a plug and play, without repeating the range dates logic. For example, the `DOBInput`, despite the probably bad naming, can be reused everywhere, and in situations where we cannot have DOB less than 18 years old (e.g. cannot be a minor) one just flags that as in `<DOBInput preventMinor />` and the component knows what that means.

Similarly, for `<IssueDateInput />` perhaps one reasonable behavior would be to by default disable any present + future dates since issue dates in our cases have to be always in the past.

And for `<ExpirationDateInput />` by default would have dates in the past disabled in the calendar.

With that, one can reuse these abstractions in Nitro without worrying about what the logic should be to disable dates within the calender.

Reply:
I get what you mean. I think you already answered #1 above, but as to #2 this would still be useful as the base component which would be extended by the thin and simple `DOBInput`, `IssueDateInput` and `ExpirationDateInput` components which I can imagine would work best if placed in `nitro_react`. 

Generally speaking, range is important to have even if we don't directly need both dates currently (we will). So looks like all I should add now is some logic to handle the other date prop not being present.

Reply:
Makes sense, that sounds good to me 👍 Also, feel free to rename `DOBInput` to something that makes more sense.